.github/scripts/code_style.sh

```diff
@@ -0,0 +1,62 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+set -o errexit # Exit if command failed.
+set -o nounset # Exit if variable not set.
+set -o pipefail # Exit if pipe failed.
+
+usage() {
+    echo "${0} [check|fix]" 1>&2
+    exit 1
+}
+
+FIX=${1:-""}
+
+# Version 0.67 is the source of truth
+if ! [ -x "$(command -v uncrustify)" ]; then
+    tmp_dir=$(mktemp -d -t uncrustify-XXXX)
+    pushd $tmp_dir
+    git clone https://github.com/uncrustify/uncrustify.git
+    cd uncrustify
+    git checkout uncrustify-0.67
+    mkdir build
+    cd build
+    cmake ..
+    cmake --build .
+    sudo make install
+    popd
+fi
+
+if [[ "$FIX" == "check" ]]; then
+    RESULT=$(uncrustify -c ./uncrustify.cfg --check           \
+    ./source/*.c                                              \
+    ./source/include/*.h                                      \
+    ./source/interface/*.h                                    \
+    ./ports/coreMQTT/*.c                                      \
+    ./ports/coreMQTT/*.h                                      \
+    ./tests/config_files/*.h                                  \
+    ./tests/e2e/device/*.c                                    \
+    ./tests/e2e/device/*.h                                    \
+    ./tests/ut/*.h                                            \
+    ./tests/ut/*.c)
+
+    if [ $? -ne 0 ]; then
+      echo $RESULT | grep "FAIL"
```

Comment:
I am assuming  uncrustify will return $? as non-zero value when failed. In that case lets exit with if we see that and fail the script.

exit $?

Reply:
The grep step attempts to parse through the large amount of text which it puts to the console to highlight what failed. I can remove it if necessary but I think it can be useful

Reply:
Weird 
set -o errexit # Exit if command failed.

should never allow grep to be called.